### PR TITLE
DEV-1385 expose prometheus metrics for catalog indexing

### PR DESCRIPTION
**Pull request by moseshll**
- Add `push_metrics` to services.
- See README for ENV variables of interest.
- Add support for checking the pushgateway for metrics in the appropriate indexer tests.
- Add a `postflight` method for the indexer command, all it does right now is finalize the pushgateway job.
- Randomly TIDY the Gemfile and docker-compose with some list alphabetization.

As mentioned in a ht_tanka PR, `JOB_NAME` is set to `index_catalog` in this branch but for consistency should be set to `catalog_indexing`, I think?

### CODE REVIEW

**Review by aelkiss — APPROVED** [review]
This all seems pleasantly straightforward especially in terms of the integration with `each_record`.

I'd be a bit inclined to default `JOB_NAME` to include the specific `cictl` subcommand being used -- the scenario I'm thinking about is wanting to ensure that ad-hoc runs don't stomp on or confuse the metrics from any more regular run.

File: docker-compose.yml [hunk 1]
```diff
@@ -56,5 +58,13 @@ services:
       <<: *healthcheck-defaults
       test: [ "CMD", "healthcheck.sh", "--su-mysql", "--connect", "--innodb_initialized" ]
 
+  pushgateway:
+    image: prom/pushgateway
+    ports:
```
[the remark this answers is not in the document]
Comment (moseshll) [reply docker-compose.yml:63]: It works fine if the ports are not exposed, but frankly I found it useful to have it available in a browser while debugging the test integration in particular.

**Comment by moseshll** [comment]
@aelkiss  Requesting a quick final once-over because I moved some things around to get at the `cictl ...` command via Thor `current_command_chain`. It de-clutters Services a little so I reckon that is okay.

**Review by aelkiss — APPROVED** [review]
All makes sense to me. We'll give it a try in the testing workflow.

**Pull request merged**